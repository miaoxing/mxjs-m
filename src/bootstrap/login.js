module.exports = require('./login.' + (process.env.TARO_ENV || 'weapp'));
