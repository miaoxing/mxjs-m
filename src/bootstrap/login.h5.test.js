import '@mxjs/taro';
import login from './login.h5';
import Taro from '@tarojs/taro';
import $, {Ret} from 'miaoxing';
import {waitFor} from '@testing-library/react';

describe('login h5', () => {
  const originalLocation = window.location;

  beforeEach(() => {
    delete window.location;
    window.location = {
      pathname: '',
      href: '',
    };
  });

  afterEach(() => {
    window.location = originalLocation;
  });

  test('suc', async () => {
    process.env.TARO_ENV = 'h5';

    window.location.href = 'https://test.com/path?a=b';
    window.location.search = '?a=b';
    window.location.pathname = '/path';

    let userAgent = jest.spyOn(window.navigator, 'userAgent', 'get');
    userAgent.mockReturnValue('MicroMessenger');

    Taro.getStorageSync = jest.fn().mockReturnValue(null);

    $.req = jest.fn().mockReturnValueOnce('test-code')
      .mockReturnValueOnce('test-state');

    $.http = jest.fn().mockResolvedValueOnce({
      ret: Ret.suc({token: 'test-token'}),
    });

    Taro.setStorageSync = jest.fn();
    Taro.redirectTo = jest.fn();

    await login();
    await waitFor(() => {
      expect(Taro.redirectTo).toBeCalled();
    });

    expect(Taro.getStorageSync).toMatchSnapshot();
    expect($.req).toMatchSnapshot();
    expect($.http).toMatchSnapshot();
    expect(Taro.setStorageSync).toMatchSnapshot();
    expect(Taro.redirectTo).toMatchSnapshot();
  });
});
