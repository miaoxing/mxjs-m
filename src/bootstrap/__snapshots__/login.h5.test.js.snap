// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login h5 suc 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "token",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`login h5 suc 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "code",
    ],
    Array [
      "state",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": "test-code",
    },
    Object {
      "type": "return",
      "value": "test-state",
    },
  ],
}
`;

exports[`login h5 suc 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "data": Object {
          "code": "test-code",
          "state": "test-state",
          "url": "https://test.com/path?a=b",
        },
        "method": "post",
        "url": "/m-api/wechat-oa/login",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`login h5 suc 4`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "token",
      "test-token",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`login h5 suc 5`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/path?a=b",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
